Add RViz Marker DisplayTypes Tutorial

LeweC opened the pull request:
This PR is a continuation of the initial issue #3864, the discussion in the abandoned PR #3908 which led to the first completed PR #4040 of the porting of old RViz tutorial series.

This is the next tutorial about RViz and especially RViz Marker Types, which is the basis for the following tutorial series I want to port in later PRs as soon as this one is ready because I have some time now.

The [link](https://wiki.ros.org/rviz/DisplayTypes/Marker) to the original Marker DisplayTypes Tutorial in ROS1 which served as a basis.

In the [old RViz documentation page](https://wiki.ros.org/rviz) the DisplayTypes: Marker was grouped with the other DisplayTypes on a [subpage](https://wiki.ros.org/rviz/DisplayTypes). In contrast, I decided to keep it on the main RViz subpage for this version, as I do not intend to add new PRs for the other DisplayTypes in the near future. I want to finish the current tutorial series first. I do not think we need a subpage for the Marker DisplayTypes page until we have more than one other DisplayType Page, and I think this page still fits nicely into the main RViz page at the moment, but I am interested in your opinion.

I think this version is ready for your next review step (as it had one or two review steps in #3908)

LeweC committed:
Add MarkerDisplayTypes

LeweC committed:
small fixes

LeweC committed:
Merge branch 'ros2:rolling' into rolling

LeweC committed:
small changes

LeweC committed:
fixes for Linter/Doc8

LeweC committed:
fix Title underline too short

ahcorde commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+* ``mesh_resource``:
+
+    The resource location for the ``MESH_RESOURCE`` marker type. Can be any mesh type supported by RViz (.stl or Ogre .mesh in 1.0, with the addition of COLLADA in 1.1).
+    The format is the URI-form used by resource_retriever, including the package:// syntax.
```
link to `resource_retriever` ?

ahcorde commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+.. image:: images/SphereListMarker.png
+
+A sphere list is a list of spheres with all the same properties except their positions.
+Using this object type instead of a `visualization_msgs/msg/MarkerArray <https://github.com/ros2/common_interfaces/blob/{DISTRO}/visualization_msgs/msg/MarkerArray.msg>`_ allows rviz to batch-up rendering,
```
```suggestion
Using this object type instead of a `visualization_msgs/msg/MarkerArray <https://github.com/ros2/common_interfaces/blob/{DISTRO}/visualization_msgs/msg/MarkerArray.msg>`_ allows RViz to batch-up rendering,
```

ahcorde commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+
+Uses the ``mesh_resource`` field in the marker.
+Can be any mesh type supported by RViz (binary .stl or Ogre .mesh in 1.0, with the addition of COLLADA (.dae) in 1.1).
+The format is the URI-form used by resource_retriever, including the ``package://`` syntax.
```
link to `resource_retriever`?

ahcorde commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+.. image:: images/mesh_resource_marker.png
+
+Uses the ``mesh_resource`` field in the marker.
+Can be any mesh type supported by RViz (binary .stl or Ogre .mesh in 1.0, with the addition of COLLADA (.dae) in 1.1).
```
```suggestion
Can be any mesh type supported by RViz (binary ``.stl`` or Ogre ``.mesh`` in 1.0, with the addition of COLLADA (.dae) in 1.1).
```

ahcorde commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+
+* ``mesh_resource``:
+
+    The resource location for the ``MESH_RESOURCE`` marker type. Can be any mesh type supported by RViz (.stl or Ogre .mesh in 1.0, with the addition of COLLADA in 1.1).
```
```suggestion
    The resource location for the ``MESH_RESOURCE`` marker type. Can be any mesh type supported by RViz (``.stl`` or Ogre ``.mesh`` in 1.0, with the addition of COLLADA in 1.1).
```

LeweC committed:
Review changes 1

ahcorde commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+
+* ``scale``:
+
+    Scale of the marker. Applied before the position/orientation. A scale of [1,1,1] means the object will be 1m by 1m by 1m.
```
```suggestion
    Scale of the marker. Applied before the position/orientation. A scale of [1, 1, 1] means the object will be 1m by 1m by 1m.
```

ahcorde commented:
@clalancette Do you mind to take a quick look ?

kscottz commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+------------------
+1 Example Usage (C++)
+^^^^^^^^^^^^^^^^^^^^^
+First we need to create a publisher and set a topic:
```
```suggestion
First we will create a simple publisher node that publishes ``Marker`` messages from the ``visualization_messages`` package to the ``visualization_marker`` topic:
```

kscottz commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+
+2 Message Parameters
+^^^^^^^^^^^^^^^^^^^^
+* ``ns``:
```
```suggestion

The Marker message type is defined in `ROS 2 Common Interfaces <https://github.com/ros2/common_interfaces/tree/{DISTRO}/visualization_msgs/msg>_` package.
The messages in this package include comments that are helpful in understanding each of the fields in the message.

* ``ns``:
```
Where possible we should point users to message definitions, especially if they are reasonably documented.
Unfortunately the docs.ros.org doesn't include the actual message definitions. Github is the next best thing. We may want to use a macro for the ROS version in the linl.

kscottz commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+
+* ``color``:
+
+    Color of the object, specified as r/g/b/a, with values in the range of [0, 1]. Don't forget to set a or it will default to 0 and be invisible.
```
```suggestion
    Color of the object, specified as r/g/b/a, with values in the range of [0, 1]. The, `a` or alpha value, denotes the opacity of the marker with 1 indicating opaque and 0 indicating completely transparent. The default value is 0, or completely transparent. **You must set the a value of your marker to a non-zero value or it will be transparent by default!**
```

kscottz commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+* ``points``:
+
+    Only used for markers of type ``Points``, ``Line strips``, and ``Line`` / ``Cube`` / ``Sphere`` -lists.
+    It's also used for the Arrow type, if you want to specify the arrow start and end points.
```
```suggestion
    It's also used for the Arrow type, if you want to specify the arrow start and end points.
    This entry represents a list of `geometry_msgs/Point` types for the center or each marker object you would like rendered.  
```

kscottz commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+
+* ``colors``:
+
+    Only used for markers that use the points member, specifies per-vertex color (no alpha yet).
```
```suggestion
    This field is only used for markers that use the points member. This field specifies per-vertex color r/g/b/ color  (no alpha yet) for each entry in `points`.
```

kscottz commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+
+* ``lifetime``:
+
+    A duration value used to automatically delete the marker after this period of time.
```
```suggestion
    A `duration message value <https://docs.ros.org/en/ros2_packages/{DISTRO}/api/builtin_interfaces/interfaces/msg/Duration.html>`_ used to automatically delete the marker after this period of time.
```
Make the user's life easier. =)

kscottz commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+
+* ``frame_locked``:
+
+    Tells RViz to retransform the marker into the current location of the specified frame every update cycle.
```
Not your fault, but I wish this was a bit clearer.  The docs are pretty sparse.

LeweC replied:
I changed it a bit. Mainly added a sentence, I hope this makes it clearer.

kscottz commented on source/Tutorials/Intermediate/RViz/Marker-Display-types/Marker-Display-types.rst:
```diff
@@ -0,0 +1,297 @@
+.. image:: images/text_view_facing_marker.png
+
+This marker displays text in a 3D spot in the world.
+The text always appears oriented correctly to the view. Uses the ``text`` field in the marker.
```
```suggestion
The text always appears oriented correctly for the RViZ user to see the included text. Uses the ``text`` field in the marker.
```

kscottz reviewed (approved):
@LeweC This is fantastic and we really appreciate your contribution! Thank you so much.

This is a fantastic start to a decent RViz marker tutorial. Obviously there is a lot more we could add here, but I think this is good enough to add in once the suggestions are merged.

LeweC committed:
Apply suggestions from code review

Co-authored-by: Katherine Scott <katherineAScott@gmail.com>
Co-authored-by: Alejandro Hernández Cordero <ahcorde@gmail.com>
Signed-off-by: Lewe Christiansen <51883655+LeweC@users.noreply.github.com>

ahcorde reviewed (changes requested):
Do you mind to check linters ? https://github.com/ros2/ros2_documentation/actions/runs/11952606475/job/33318689926?pr=4821

LeweC committed:
review changes and linter

LeweC commented:
> Do you mind to check linters ? https://github.com/ros2/ros2_documentation/actions/runs/11952606475/job/33318689926?pr=4821

Yeah I was just in the procces of doing that. I am pretty sure my setup is/was correct, I think it came from one of the suggestions. Fixed now and I think I added all your guys feedback.

ahcorde reviewed (approved):
Thank you for iterating with us!
